File: src/mod/applications/mod_vmd/scripts/vmd.js
function onInput(s, type, obj, arg)
{
	try
	{
		if(type == "dtmf")
		{
			console_log("info", "DTMF digit: "+s.name+" ["+obj.digit+"] len ["+obj.duration+"]\n");
		}
		else if(type == "event" && session.getVariable("vmd_detect") == "TRUE")
		{
			console_log("info", "Voicemail Detected\n");
		}
		
	}
	catch(e)
	{
		console_log("err", e + "\n");
	}
	return true;
}

session.answer();
session.execute("vmd", "start");
while(session.ready())
{
	session.streamFile(argv[0], onInput);
}

File: src/mod/applications/mod_vmd/scripts/vmd.lua
local human_detected = false;
local voicemail_detected = false;

function onInput(session, type, obj)
    if type == "dtmf" and obj['digit'] == '1' and human_detected == false then
        human_detected = true;
        return "break";
    end

    if type == "event" and voicemail_detected == false then
        voicemail_detected = true;
        return "break";
    end
end

session:setInputCallback("onInput");
session:execute("vmd","start");
